### PR TITLE
Send requests with human User-Agent header to reduce false positives

### PR DESCRIPTION
**Pull request by jakejarvis**
There might be a better way to do this, but after some quick testing I've found specifying a User-Agent header that resembles a human (i.e. Chrome on macOS) drastically cut down on non-404 errors (401, 403, 502, etc.).

I added this quickly without much familiarity with the code (did run `test` and `fix`, though) so feel free to make/suggest a better implementation. Maybe it's worth making the specific agent string customizable via an option/flag too, but I can't think of a scenario where that would really be necessary?

Thanks! :)

### CODE REVIEW

**Comment by pjoulot** [comment]
Yeah, it reduces the false positives a lot. I think we should add that as an argument in order to allow people to use whatever user-agent they want.

**Comment by JustinBeckwith** [comment]
I love the idea, but I'm just a little bit scared that it's going to get folks IPs banned :) Usually not respecting robots.txt, and faking out the user agent can get you into a little trouble.  I'm not saying no, just saying I probably need to do a bit of research.

**Comment by pjoulot** [comment]
If we add it as an argument, it gives the choice to the people to use it or not.
Honestly so many websites return an error code when not using a fake user-agent, I definitely need something otherwise the results of linkinator are half wrong.

File: src/index.ts [hunk 1]
```diff
@@ -181,6 +181,10 @@ export class LinkChecker extends EventEmitter {
       let res = await gaxios.request<string>({
         method: opts.crawl ? 'GET' : 'HEAD',
         url: opts.url.href,
+        headers: {
```
Comment (JustinBeckwith) [inline src/index.ts]: Let's define this once, globally, and then use it in both places where we need it.

**Comment by JustinBeckwith** [comment]
👋 @jakejarvis this is something I'd love to land!  Any way I could trouble you for a test?

**Comment by jakejarvis** [comment]
@JustinBeckwith Of course, really sorry for neglecting this! Will re-visit today 😊

**Review by JustinBeckwith — APPROVED** [review]

**Pull request merged**